org-scan/gitleaks.toml

```diff
@@ -145,42 +170,75 @@ title = "gitleaks config"
 	regex = '''(?i)twilio(.{0,20})?['\"][0-9a-f]{32}['\"]'''
 	tags = ["key", "twilio"]
 
-[[rules]]
-	description = "Env Var"
-	regex = '''(?i)(apikey|secret|key|api|password|pass|pw|host)=[0-9a-zA-Z-_.{}]{4,120}'''
-
 [[rules]]
 	description = "High Entropy"
 	regex = '''[0-9a-zA-Z-_!{}/=]{4,120}'''
-  	fileNameRegex = '''(?i)(dump.sql|high-entropy-misc.txt)$'''
+ 	fileNameRegex = '''(?i)(dump.sql|high-entropy-misc.txt)$'''
 	tags = ["entropy"]
-        [[rules.Entropies]]
-            Min = "4.3"
-            Max = "7.0"
-        [[rules.whitelist]]
-            description = "ignore ssh key and pems"
-            file = '''(pem|ppk|env)$'''
-            path = '''(.*)?ssh'''
-
-[[rules]]
-	description = "Potential bash var"
-	regex='''(?i)(=)([0-9a-zA-Z-_!{}=]{4,120})'''
-	tags = ["key", "bash", "API", "generic"]
-        [[rules.Entropies]]
-            Min = "3.5"
-            Max = "4.5"
-            Group = "1"
+	[[rules.Entropies]]
+			Min = "4.3"
+			Max = "7.0"
+	[[rules.whitelist]]
+		description = "ignore ssh key and pems"
+		file = '''(pem|ppk|env)$'''
+		path = '''(.*)?ssh'''
 
 [[rules]]
 	description = "WP-Config"
 	regex='''define(.{0,20})?(DB_CHARSET|NONCE_SALT|LOGGED_IN_SALT|AUTH_SALT|NONCE_KEY|DB_HOST|DB_PASSWORD|AUTH_KEY|SECURE_AUTH_KEY|LOGGED_IN_KEY|DB_NAME|DB_USER)(.{0,20})?['|"].{10,120}['|"]'''
 	tags = ["key", "API", "generic"]
 
 [[rules]]
-	description = "Files with keys and credentials"
-    fileNameRegex = '''(?i)(id_rsa|passwd|id_rsa.pub|pgpass|pem|key|shadow)'''
+	description = "yaml secrets"
+	regex = '''(?i)(password|enc.key|auth.pass|secret\w+):\s+(.*)'''
+	fileNameRegex = '''(?i)(\.yml|\.yaml)'''
+	tags = ["yaml"]
+	[[rules.whitelist]]
+		regex = '''(\(\(.*\)\)|\{\{.*\}\})'''
+		description = "ignore substituted values"
+	[[rules.whitelist]]
+		description = "ignore explicit example and test files"
+		file = "example.yml"
```

Comment:
I'm not actually sure about this - I think one way examples are often made is to copy a running config and sub out the credentials, but what if you forget to replace the credentials?

I don't have a better idea, and examples are the place where caulking trips me up the most, so maybe this is the sanest option?

Reply:
This is a retrospective observation, that  a lot of things that looked like secrets were in files named `example.yml`. This pattern (and this config file) is only used for scanning of repositories that already exist, and at one point I was trying to get that retrospective scan to come out with zero matches. That was a fool's errands, and I've instead determined that all patterns matching before today we're false positives or non-applicable. 

I could remove some of these exceptions, but I think making this tighter is an effort of diminishing returns.